crates/oro-client/src/client.rs

```diff
@@ -8,14 +8,56 @@ use reqwest::Client;
 #[cfg(not(target_arch = "wasm32"))]
 use reqwest::ClientBuilder;
 #[cfg(not(target_arch = "wasm32"))]
+use reqwest::{NoProxy, Proxy};
+#[cfg(not(target_arch = "wasm32"))]
 use reqwest_middleware::ClientWithMiddleware;
 use url::Url;
 
+#[cfg(not(target_arch = "wasm32"))]
+use crate::OroClientError;
+
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct OroClientProxyConfig {
+    pub proxy: bool,
+    pub proxy_url: Option<String>,
```

Comment:
```suggestion
    pub proxy_url: Option<Proxy>,
```

Maybe this should take an actual `Proxy`, and have the builder method build that? That way, we can run the validations we need before `.build()`